src/groupByKey.ts

```diff
@@ -2,6 +2,13 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((result, item) => {
+    const keyValue = String(item[key]);
+
+    return {
+      ...result,
+      [keyValue]: [...(result[keyValue] || []), item],
```

Comment:
The spread operator is used inside a loop, which can lead to performance issues due to creating a new object on each iteration. Consider using a different approach to add properties to the 'result' object.

We can just simplify this logic, please remove your `return {}` and try this one to be:

```
if (!result[keyValue]) {
   result[keyValue] = [];
}
result[keyValue].push(item);

return result;
```